### PR TITLE
Pablo/process game

### PR DESCRIPTION
**Pull request by Pabloj20**
Check if all the handle_call/cast are correct and if the functions I have used (for example insert_boat) have the correct arguments

### CODE REVIEW

**Review by RodrigoMolinaMadrid — CHANGES_REQUESTED** [review]
Change the things that we said, in the lines 18, 49, 50

**Comment by nurietzsche2021** [comment]
Pablo, I would like to have better specs, actually, They look large and messy because you haven't set your specific types globally. Please, do that like I have in my file, defining them on the top of the file. Then, you can use them and they are recognisable by name.

**Comment by nurietzsche2021** [comment]
I left all the comments on the pdf I sent you but, basically I think function player is useless and you could have used a map or a keywordlist in order to avoid invoking this function, you could just pass me the key from the value. I insist on the usage of our specific types in documentation in order to help our understanding. Lastly but most importantly, I insist on the ERROR topic. I  haven't seen any error handling and informing the user, you should ask me the proper functions to check the possible errors. 

To sum up, we must discuss deeply next monday how we are going to approach every step of the game because I need specific and clear instructions to build my functions efficiently.
